services/platform/platform.go

```diff
@@ -110,16 +115,78 @@ func (s *Service) Connect(ctx context.Context, req *platformpb.ConnectRequest) (
 	})
 	if err != nil {
 		s.l.Errorf("Failed to insert SSO details: %s", err)
-		return nil, status.Error(codes.Internal, "Internal server error")
+		return nil, status.Error(codes.Internal, internalServerError)
 	}
 
 	if err := s.UpdateSupervisordConfigurations(ctx); err != nil {
 		s.l.Errorf("Failed to update configuration of grafana after connecting PMM to Portal: %s", err)
-		return nil, status.Error(codes.Internal, "Internal server error")
+		return nil, status.Error(codes.Internal, internalServerError)
 	}
 	return &platformpb.ConnectResponse{}, nil
 }
 
+// Disconnect disconnects a PMM server from the organization created on Percona Portal.
+func (s *Service) Disconnect(ctx context.Context, req *platformpb.DisconnectRequest) (*platformpb.DisconnectResponse, error) {
+	ssoDetails, err := models.GetPerconaSSODetails(ctx, s.db.Querier)
+	if err != nil {
+		return nil, status.Error(codes.Aborted, "PMM server is not connected to Portal")
+	}
+
+	settings, err := models.GetSettings(s.db)
+	if err != nil {
+		s.l.Errorf("Failed to fetch PMM server ID and address: %s", err)
+		return nil, status.Error(codes.Internal, internalServerError)
+	}
+
+	if err := s.CleanSupervisordConfigurations(ctx); err != nil {
```

Comment:
i don't see the need for a separate method for this, `CleanSupervisordConfiguration` calls `models.GetSettings` (which has already been called above)  and then `s.supervisord.UpdateConfiguration(settings, nil)`  why not just call `s.supervisord.UpdateConfiguration` here instead of a having a separate method, i don't see it being reused anywhere else.

Reply:
Yes right.